### ISSUE TITLE
Monitoring Fail For region The operation couldn’t be completed. (kCLErrorDomain error 5.)

### ISSUE THREAD

**Issue by summer9011** [mobiquity-networks/MNNotificationsManager-iOS-SDK#6]
I called `[MNNotificationsManager start];` method，it cause this problem，

Monitoring Fail For region 538c7d560af5509c247b23c6: The operation couldn’t be completed. (kCLErrorDomain error 5.)

Thread Call Stack: (
	0   P365PrivatePodTest                  0x000000010025a7d0 -[M2MLogUtils .cxx_destruct] + 183432
	1   CoreLocation                        0x000000018cb8ca04 CLClientGetCapabilities + 41160
	2   CoreLocation                        0x000000018cb7de7c CLClientInvalidate + 1036
	3   CoreFoundation                      0x0000000184b88160 <redacted> + 20
	4   CoreFoundation                      0x0000000184b8795c <redacted> + 288
	5   CoreFoundation                      0x0000000184b85c54 <redacted> + 1976
	6   CoreFoundation                      0x0000000184ab4048 CFRunLoopRunSpecific + 444
	7   GraphicsServices                    0x000000018653a198 GSEventRunModal + 180
	8   UIKit                               0x000000018aaa02fc <redacted> + 684
	9   UIKit                               0x000000018aa9b034 UIApplicationMain + 208
	10  P365PrivatePodTest                  0x0000000100036058 main + 124
	11  libdyld.dylib                       0x0000000183a985b8 <redacted> + 4
)

**Comment by dnapolitano** [comment]
I think you may already have discovered your issue, but just in case: please note that it's not possible to monitor for Bluetooth notifications on a simulator. Please build your project onto a physical test phone and retry. That should work fine.

**Comment by summer9011** [comment]
No,  I tried this demo on a physical phone (iPhone6, base on iOS10.1) all the time, and always received this issue when I called `[MNNotificationsManager start];`。
So I viewed `CLLocationManager` class in  the Apple Documentation, found out `- (void)startMonitoringForRegion:(CLRegion *)region` can cause the issue. Apple told that "An app can  register up  to 20 regions at a time.".  
I suppose that the SDK contains this method and to moniting more than 20 regions when I called start method. If It's truth, should you create the API to let me how to control the case or clear the case in the SDK.

**Comment by dnapolitano** [comment]
We monitor for 3 Bluetooth and 3 geofence regions, 6 total. If you have additional SDKs introducing a sufficient number of additional regions that the app is exceeding 20, you would need to implement an app-level region management strategy.

**Comment by dnapolitano** [comment]
May I ask, please: did you integrate our SDK manually, or did you use CocoaPods? Thanks!

**Comment by dnapolitano** [comment]
By the way, you are also welcome to email me at: dan@mobiquitynetworks.com. If it turns out your app is (by virtue of our 6 additional regions) now over the 20 limit, if you need, I can send you some notes about how to implement region management. If that's not the case and something else is happening, it may be easier for us to collaborate on this via email--but I will keep an eye on this item and try to be responsive to you wherever you inquire. Thanks again.

**Comment by summer9011** [comment]
Hey, thank you for attention. this days, I created few demos to testing this issue. By the way, I integrated your SDK using CocoaPods and also integrated manually. So If the demo only contains your SDK, it's running normally, but contains another SDK called BeaconsInSpace, it can cause this issue when I call `[MNNotificationsManager start];`. so maybe something wrong in SDK, would you please to check the SDK, thank you again. I will email to you the demo contains 2 SDKs, this demo using CocoaPods.

**Comment by dnapolitano** [comment]
How many regions is BeaconsInSpace monitoring?

Sincerely,

Dan
